release/src/Util.hs

```diff
@@ -233,32 +245,64 @@ artifactFiles allArtifacts art@Artifact{..} = do
+
+    let shouldReleasePlatInd = shouldRelease allArtifacts (PlatformDependent False)
 
     pure $
         [(directory </> mainArtifactIn, outDir </> mainArtifactOut) | shouldRelease allArtifacts artPlatformDependent] <>
-        [(directory </> pomFileIn, outDir </> pomFileOut) | isJar artReleaseType, shouldRelease allArtifacts (PlatformDependent False)] <>
-        [(directory </> sourceJarIn, outDir </> sourceJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just sourceJarIn <- pure mbSourceJarIn] <>
-        [(directory </> scalaSourceJarIn, outDir </> scalaSourceJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just scalaSourceJarIn <- pure mbScalaSourceJarIn] <>
-        [(directory </> javadocJarIn, outDir </> javadocJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just javadocJarIn <- pure mbJavadocJarIn]
+        [(directory </> pomFileIn, outDir </> pomFileOut) | isJar artReleaseType, shouldReleasePlatInd] <>
+        [(directory </> sourceJarIn, outDir </> sourceJarOut) | shouldReleasePlatInd, Just sourceJarIn <- pure mbSourceJarIn] <>
+        [(directory </> scalaSourceJarIn, outDir </> scalaSourceJarOut) | shouldReleasePlatInd, Just scalaSourceJarIn <- pure mbScalaSourceJarIn] <>
+        [(directory </> javadocJarIn, outDir </> javadocJarOut) | shouldReleasePlatInd, Just javadocJarIn <- pure mbJavadocJarIn] <>
+        [(directory </> scaladocJarIn, outDir </> scaladocJarOut) | shouldReleasePlatInd, Just scaladocJarIn <- pure mbScaladocJarIn]
+
+-- | Given an artifact, produce a list of pairs of an input file and the Maven coordinates
+artifactCoords :: E.MonadThrow m => AllArtifacts -> Artifact PomData -> m [(MavenCoords, Path Rel File)]
+artifactCoords allArtifacts art@Artifact{..} = do
+    let PomData{..} = artMetadata
+    let (directory, name) = splitBazelTarget artTarget
+    let jarClassifier =  if getPlatformDependent artPlatformDependent then Just osName else Nothing
+    directory <- parseRelDir $ unpack directory
+
+    mainArtifactFile <- parseRelFile $ unpack $ mainFileName artReleaseType name
+    pomFile <- parseRelFile (unpack (name <> "_pom.xml"))
+    mbSourceJarFile <- traverse (parseRelFile . unpack) (sourceJarName art)
+    mbScalaSourceJarFile <- traverse (parseRelFile . unpack) (scalaSourceJarName art)
+    mbJavadocJarFile <- traverse (parseRelFile . unpack) (javadocJarName art)
+    mbScaladocJarFile <- traverse (parseRelFile . unpack) (scaladocJarName art)
+
+    let mavenCoords classifier artifactType =
+           MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier, artifactType }
+    let shouldReleasePlatInd = shouldRelease allArtifacts (PlatformDependent False)
+
+    pure $ [ (mavenCoords jarClassifier $ mainExt artReleaseType, directory </> mainArtifactFile) | shouldReleasePlatInd] <>
```

Comment:
Not sure if `shouldReleasePlatInd` should appear on this line and the next (the pom file).

Reply:
Good catch! Too much copy and pasting...

Reply:
implemented In 2d9b593a6